Tests/UserServiceTests/Validators/UserEmailPasswordValidatorTests.cs

```diff
@@ -0,0 +1,283 @@
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstNameContainsNumbers()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "Leo12",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameError);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstNameFirstLetterIsNotUpperCase()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameError);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstNameRestLettersIsNotLowerCase()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "LeO",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameError);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void WrongBirthday()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Now.AddDays(1)
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FutureErrorBirthday);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissBirthday()
```

Comment:
Why do you check miss only with birthday? What if we miss name or surname?

Reply:
add tests